src/traversal/mod.rs

```diff
@@ -284,22 +285,26 @@ pub fn descendants(graph: &digraph::PyDiGraph, node: usize) -> HashSet<usize> {
 ///     graph are searched.
 /// :param visitor: A visitor object that is invoked at the event points inside the
 ///     algorithm. This should be a subclass of :class:`~rustworkx.visit.BFSVisitor`.
+///     This has a default value of ``None`` as a backwards compatibility artifact (to
+///     preserve argument ordering from an earlier version) but it is a required argument
+///     and will raise a ``TypeError`` if not specified.
 #[pyfunction]
-#[pyo3(text_signature = "(graph, source, visitor)")]
 pub fn digraph_bfs_search(
     py: Python,
     graph: &digraph::PyDiGraph,
     source: Option<Vec<usize>>,
-    visitor: PyBfsVisitor,
+    visitor: Option<PyBfsVisitor>,
 ) -> PyResult<()> {
+    if visitor.is_none() {
+        return Err(PyTypeError::new_err("Missing required argument visitor"));
+    }
```

Comment:
I think this is the only problematic part of the PR... Why can't `digraph_bfs_search` accept `PyBfsVisitor`? I think PyO3 should check that the argument is valid, not us

Reply:
Basically we had two options to fix this, either give visitor a default option or move `visitor` before `source`. I opted to give it a default here because moving `visitor` would be a breaking change because it would break existing callers of this function using positional arguments

Reply:
Well, I guess we’ll have to deal with it